controllers/liveupdate_controller.go

```diff
@@ -0,0 +1,653 @@
+	if selector["version"] != "" {
+		_, err := controllerutil.CreateOrPatch(r.ctx, r, sourceClusterService, func() error {
+			if ispn.CreationTimestamp.IsZero() {
+				return errors.NewNotFound(schema.ParseGroupResource("infinispan.infinispan.org"), ispn.Name)
+			}
+			delete(selector, "version")
+			sourceClusterService.Spec.Selector = selector
+			return nil
+		})
+		if err != nil {
+			r.reqLogger.Error(err, "Failed to update service")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Remove version label from the pods
+	targetStatefulSet := &appsv1.StatefulSet{}
+	err := r.Get(r.ctx, types.NamespacedName{Namespace: r.infinispan.GetNamespace(), Name: r.targetName}, targetStatefulSet)
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "Failed to find the new statefulSet")
+		return reconcile.Result{}, err
+	}
+	podList, err := r.podList(targetStatefulSet)
+	if err != nil {
+		r.reqLogger.Error(err, "Error getting pods of the new statefulset")
+		return reconcile.Result{}, err
+	}
+
+	for _, pod := range podList.Items {
+		_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, &pod, func() error {
+			if pod.CreationTimestamp.IsZero() {
+				return errors.NewNotFound(corev1.Resource(""), pod.Name)
+			}
+			labels := pod.ObjectMeta.Labels
+			delete(selector, "version")
+			pod.Labels = labels
+			return nil
+		})
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Error deleting pod '%s' version ", pod.Name)
+		}
+	}
+
+	// Store ping service name from the original cluster
+	oldPingName := ispn.GetPingServiceName()
+
+	// Change statefulSet reference in the Infinispan resource
+	_, err = controllerutil.CreateOrPatch(r.ctx, r, ispn, func() error {
+		if ispn.CreationTimestamp.IsZero() {
+			return errors.NewNotFound(schema.ParseGroupResource("infinispan.infinispan.org"), ispn.Name)
+		}
+		ispn.Status.StatefulSetName = r.targetName
+		return nil
+	})
+
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("error setting statefulSet '%s' reference to Infinispan", r.targetName)
+	}
+
+	// Delete ping service from the original cluster
+	pingService := &corev1.Service{}
+	err = r.Get(context.TODO(), types.NamespacedName{Namespace: ispn.Namespace, Name: oldPingName}, pingService)
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "source ping service  not found")
+		return reconcile.Result{}, err
+	}
+	err = r.Delete(context.TODO(), pingService)
+	if err != nil {
+		r.reqLogger.Error(err, "Error Deleting source ping service")
+		return reconcile.Result{}, err
+	}
+
+	// Delete old statefulSet
+	sourceStatefulSet := &appsv1.StatefulSet{}
+	err = r.Get(context.TODO(), types.NamespacedName{Namespace: ispn.Namespace, Name: ispn.Name}, sourceStatefulSet)
+
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "source stateful set not found")
+		return reconcile.Result{}, err
+	}
+
+	err = r.Delete(r.ctx, sourceStatefulSet)
+	if err != nil {
```

Comment:
There's no need to do both `GET` and `DELETE` here. We can just call `DELETE` and if  `errors.IsNotFound` continue.